mdazfar2/Ezyshop
# products by shop

### Description
Implemented a feature where clicking **"Load Products"** redirects to a page displaying only **products related to the selected seller**. Necessary backend and frontend changes were made to ensure proper functionality.

### Changes
- **Backend:** Modified API to fetch products by seller ID with relevant details (category, seller info).
- **Frontend:** Updated product page to display products based on seller, with loading and error handling states.

### How to Test
1. Click **"Load Products"** on a shop page.
2. Verify that the product page shows **only the selected seller’s products**.
3. Ensure the **store name** is displayed correctly.
4. Test with multiple sellers to confirm the functionality.
5. Handle empty product lists gracefully with a fallback message.

### Checklist
- [x] Backend API fetches products by seller ID.
- [x] Frontend displays seller-specific products with loading/error handling.
- [x] UI remains consistent with the current design.


https://github.com/user-attachments/assets/9fa67765-f3ca-4af6-99ac-cfc003ca06fa


https://github.com/user-attachments/assets/03d948f9-e2b4-4a6e-8581-7155a7740fd7


https://github.com/user-attachments/assets/d8340ccf-7f65-4caf-bbaf-04c99272f2af



 closes #852
